src/go/cmd/metadata_merge/metadata_merge.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/bazelbuild/bazel-toolchains/src/go/pkg/metadata"
+)
+
+var (
+	outFile = flag.String("outFile", "", "Output merged YAML file to generate.")
+)
+
+// strArgList implements a command line flag that can be specified multiple
+// times to define a list of values.
+type strArgList struct {
+	// Args is the list of command line flags.
+	Args []string
+}
+
+func (l *strArgList) String() string {
+	return fmt.Sprintf("%v", l.Args)
+}
+
+// Set appends the given value for a particular occurance of the flag to the
+// list of flag values.
+func (l *strArgList) Set(value string) error {
+	l.Args = append(l.Args, value)
+	return nil
+}
+
+// metadataYAML stores the metadata generated by security_check & download_pkgs
+// rules as a single YAML object.
+type metadataYAML struct {
+	// Tags is the list of vulnerability tags read from YAML files generated
+	// by the security_check rule.
+	Tags []string `yaml:"tags"`
+	// Packages is the list of software package entries read from YAML files
+	// generated by the download_pkgs rule.
+	Packages []metadata.PackageMetadata `yaml:"packages"`
+
+	// tagsLookup maintains a map of tags in the "Tags" field.
+	tagsLookup map[string]bool
+}
+
+// merge merges the contents of the metadataYaml 'from' into the metadataYAML
+// 'm'. This does the following:
+// 1. Add every tag that appears in 'from' into 'm' if it doesn't already exist
+//    in 'm'.
+// 2. Add every package that apppears in 'from' into 'm'.
```

Comment:
what happens if a package already existed previously (at either a higher or lower version)?

Reply:
This binary will used to merge the security & package metadata for a specific tarball of distinct debian packages. It is not expected to have a situation where multiple packages of the same name and differing versions exist in the input YAMLs. If they exist, both entries for the same package will be added to the YAML but I don't think it's worth adding the complexity to check that in this binary just yet.